mono/metadata/marshal-ilgen.c

```diff
@@ -4808,7 +4838,7 @@ emit_marshal_vtype_ilgen (EmitMarshalContext *m, int argnum, MonoType *t,
 
 			if (!((t->attrs & PARAM_ATTRIBUTE_IN) && !(t->attrs & PARAM_ATTRIBUTE_OUT))) {
 				if (!from_oadate)
-					from_oadate = get_method_nofail (date_time_class, "FromOADate", 1, 0);
+					mono_atomic_store_seq (&from_oadate, get_method_nofail (date_time_class, "FromOADate", 1, 0));
```

Comment:
Same with this one.  the scope of `from_oadata` could be narrowed to just the `if (!((t->attrs & PARAM_ATTRIBUTE_IN) && ...)` and then could be inited with `MONO_STATIC_POINTER_INIT`, couldn't it?